subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c

```diff
@@ -255,6 +255,11 @@ img_mgmt_state_read(struct smp_streamer *ctxt)
 	     zcbor_list_start_encode(zse, 2 * CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER);
 
 	for (i = 0; ok && i < 2 * CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER; i++) {
+		/* Slot 2 mcuboot_primary_1 only exists when ram_flash is enabled*/
+		if (i == 2 && !IS_ENABLED(CONFIG_FLASH_SIMULATOR)) {
+			continue;
```

Comment:
This is not acceptable, slot 2 does not mean it is an nrf5340, it could be used on any number of devices and is being considered for the nrf9160 with MCUmgr client support

Reply:
I think this PR can go upstream. The easiest fix is to do the ERR->WRN, but I am not familiar enough in this territory to suggest something better off the top of my head.

Reply:
It is an error if someone tries to open a flash area that doesn't exist, so replacing it with a warning doesn't make sense

Reply:
Maybe img_mgmt_state_read should not read an unsexisted partition.
Is there any way to prevent this?

Reply:
Updatable image numbers should not be set to 2 for the application if it can't use both updatable images, that'd be a bug in the application configuration.

Reply:
That's what Nordic designed for nRF5340.
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/working_with_nrf/nrf53/nrf5340.html#simultaneous-multi-image-dfu

> Note
The application core does not have direct access to the network core flash memory. The update image is passed indirectly using RAM. Because of this, the mcuboot_primary_1 must be stored in ram_flash region. To enable providing such region on the device, see [CONFIG_FLASH_SIMULATOR](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/kconfig/index.html#CONFIG_FLASH_SIMULATOR).

Reply:
It might be used by nrf5340 for network core update but that does not mean that is the only use for it, I am the upstream mcumgr maintainer for zephyr.